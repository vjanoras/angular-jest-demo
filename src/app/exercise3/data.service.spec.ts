import { DataService } from "./data.service";

describe('DataService', () => {
    let service;

    const http = {
      get: jest.fn() };

    
    const listData =  [ {
        "id": 1,
        "name": "test 1"
    },
    {
      "id": 2,
      "name": "test 2"
    }];

    const stringData = "hello world";

    beforeEach(() => {
      service = new DataService(http as any);
    });
  
    it('Can return list data from API', () => {
        
        http.get.mockReturnValue(listData);
        expect(service.get("test/api/get")).toBe(listData);
    });

    it('Should match snapshot data', () => {
        
        http.get.mockReturnValue(listData);
        expect(service.get("test/api/get")).toMatchSnapshot();
    });


    it('Can return string data', () => {
        
        http.get.mockReturnValue(stringData);
        expect(service.get("test/api/get")).toBe(stringData);
    });


  });

