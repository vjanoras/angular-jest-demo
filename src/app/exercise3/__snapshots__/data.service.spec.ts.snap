// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataService Should match snapshot data 1`] = `
Array [
  Object {
    "id": 1,
    "name": "test 1",
  },
  Object {
    "id": 2,
    "name": "test 2",
  },
]
`;
